wdl2cwl/tests/cwl_files/blast.cwl

```diff
@@ -0,0 +1,335 @@
+cwlVersion: v1.2
+$graph:
+- class: Workflow
+  id: main
+  inputs:
+    - id: blast_docker_override
+      type:
+        - string
+        - 'null'
+    - id: blast_docker
+      type: string
+      # TODO: how to translate the original?
+      #       String blast_docker = select_first([blast_docker_override,"swr.cn-south-1.myhuaweicloud.com/cngbdb/blast:1.2"])
```

Comment:
:point_up:

Reply:
For the general case of programmatic conversion, we would need to synthesize a new step to compute the usable value of `blast_docker` and use the output of that step everywhere that originally referred to the `blast_docker` input.

However, if I was converting this by hand I would have combined `blast_docker_override` and `blast_docker` into a single CWL input with a default value. However, this is being used to override the docker image name with a dynamic value, which is [not yet supported in CWL](https://github.com/common-workflow-lab/wdl-cwl-translator#incompatibilities-possibly-requiring-manual-intervention); so it would be useless :-P

Anyhow, we still need a general way to convert WDL expressions in the definition of inputs. I recommend creating a separate issue for that; modifying this WDL to not need workflow input declaration expressions, and focusing on synthesizing new steps for workflow output expressions.